flink-core/src/main/java/org/apache/flink/api/common/typeutils/TypeSerializerSerializationUtil.java

```diff
@@ -200,7 +197,7 @@ public static void writeSerializersAndConfigsWithResilience(
 			for (int i = 0; i < numSerializersAndConfigSnapshots; i++) {
 
 				bufferWithPos.setPosition(offsets[i * 2]);
-				serializer = tryReadSerializer(bufferWrapper, userCodeClassLoader);
+				serializer = tryReadSerializer(bufferWrapper, userCodeClassLoader, true);
```

Comment:
This could be the place where we catch a `UnloadableSerializerException`, but if we let the caller do the iteration from 0 to `numSerializersAndConfigSnapshots`, we can push it out even more. Why is it helpful to create a list here? Otherwise we can do the exception handling in the caller and more fine grained.

Reply:
The problem is that this method might mix too many things together, that is also again visible in the complex return type and e.g. many call sites are only interested in the first element of the list. Wonder if we should break this up in dedicated steps (serializer, config) and let the callers invoke them one by one, so that we can handle exceptions on a higher level and make decisions about if we need to have a serializer there.